plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/project/FeatureDevSessionContext.kt

```diff
@@ -0,0 +1,224 @@
+            return false
+        }
+
+        // Exclude files specified by gitignore or outside the workspace:
+        if (!isWorkspaceSourceContent(file, workspaceContentRoots, changeListManager, additionalGlobalIgnoreRules)) {
+            return false
+        }
+
+        // Exclude files outside the selection root working on a subset of the workspace:
+        if (!VfsUtil.isAncestor(selectionRoot, file, false)) {
+            return false
+        }
+
+        // When auto build is enabled, include all other files:
+        return if (isAutoBuildFeatureEnabled) {
+            true
+        } else {
+            // When auto build is disabled, explicitly exclude devfile:
+            // FIXME: There should be a stronger signal to the agent than presence of the devfile in the uploaded files to enable auto build
+            !isDevFile(file)
+        }
+    }
+
+    private suspend fun zipFiles(isAutoBuildFeatureEnabled: Boolean?): File = withContext(getCoroutineBgContext()) {
+        val files = mutableListOf<VirtualFile>()
+        val ignoredExtensionMap = mutableMapOf<String, Long>().withDefault { 0L }
+        var totalSize: Long = 0
+
+        workspaceContentRoots.forEach { contentRoot ->
+            VfsUtil.visitChildrenRecursively(
+                contentRoot,
+                object : VirtualFileVisitor<Unit>(NO_FOLLOW_SYMLINKS) {
+                    override fun visitFile(file: VirtualFile): Boolean {
+                        val isIncluded = shouldIncludeInZipFile(file, isAutoBuildFeatureEnabled == true)
+                        if (!isIncluded) {
+                            val extension = file.extension.orEmpty()
+                            ignoredExtensionMap[extension] = (ignoredExtensionMap[extension] ?: 0) + 1
+                            return false
+                        }
+
+                        if (file.isFile) {
+                            totalSize += file.length
+                            files.add(file)
+
+                            if (maxProjectSizeBytes != null && totalSize > maxProjectSizeBytes) {
+                                throw RepoSizeLimitError(AwsCoreBundle.message("amazonqFeatureDev.content_length.error_text"))
+                            }
+                        }
+                        return true
+                    }
+                }
+            )
+        }
+
+        for ((key, value) in ignoredExtensionMap) {
+            AmazonqTelemetry.bundleExtensionIgnored(
+                count = value,
+                filenameExt = key
+            )
+        }
+
+        // Process files in parallel
+        val filesToIncludeFlow = channelFlow {
+            // chunk with some reasonable number because we don't actually need a new job for each file
+            files.chunked(50).forEach { chunk ->
+                launch {
+                    for (file in chunk) {
+                        send(file)
+                    }
+                }
+            }
+        }
```

Comment:
i see that this is the existing impl, but the flow seems superfluous for what is being accomplished here